Gems/Prefab/PrefabBuilder/PrefabGroup.cpp

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "PrefabGroup.h"
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/RTTI/BehaviorContext.h>
+#include <AzCore/JSON/error/error.h>
+#include <AzCore/JSON/error/en.h>
+
+namespace AZ::SceneAPI::SceneData
+{
+     PrefabGroup::PrefabGroup()
+        : m_id(Uuid::CreateNull())
+        , m_name()
+    {
+         m_prefabDom.SetNull();
+    }
+
+    const AZStd::string& PrefabGroup::GetName() const
+    {
+        return m_name;
+    }
+
+    void PrefabGroup::SetName(AZStd::string&& name)
+    {
+        m_name = AZStd::move(name);
+    }
+
+    const Uuid& PrefabGroup::GetId() const
+    {
+        return m_id;
+    }
+
+    void PrefabGroup::SetId(Uuid&& id)
+    {
+        m_id = AZStd::move(id);
+    }
+
+    Containers::RuleContainer& PrefabGroup::GetRuleContainer()
+    {
+        return m_rules;
+    }
+
+    const Containers::RuleContainer& PrefabGroup::GetRuleContainerConst() const
```

Comment:
You're not consistent with these function names, you used two different names (one with const, one without) for the rule container, but for GetSceneNodeSelectionList you don't.

Reply:
will do

Reply:
Well unfortunately these are pure abstract calls from base classes of IManifestObject

AZ.SceneAPI.DataTypes.IManifestObject
	virtual Containers::RuleContainer& GetRuleContainer() = 0;
	virtual const Containers::RuleContainer& GetRuleContainerConst() const = 0;

AZ.SceneAPI.DataTypes.IGroup
	virtual Containers::RuleContainer& GetRuleContainer() = 0;
	virtual const Containers::RuleContainer& GetRuleContainerConst() const = 0;

So if we want to clear this up we will need to properly write up a ticket to do this work since it directly affects public API that will make other APIs fail to compile that we have no access to right now.

Reply:
Ah, gotcha